src/VisualStudio/Core/Def/ProjectSystem/MetadataReferences/VisualStudioMetadataReferenceManager.cs

```diff
@@ -48,7 +50,7 @@ internal sealed partial class VisualStudioMetadataReferenceManager : IWorkspaceS
     /// <summary>
     /// Access locked with <see cref="_metadataCacheLock"/>.
     /// </summary>
-    private readonly Dictionary<FileKey, AssemblyMetadata> _metadataCache = [];
+    private readonly Dictionary<FileKey, WeakReference<AssemblyMetadata>> _metadataCache = [];
```

Comment:
💡 If you want to release the metadata as soon as nothing has a strong reference, but still have the cache, you can use `ReferenceCountedDisposable<AssemblyMetadata>.WeakReference` here. Not required, since leveraging that would require propagating RCD throughout the locations where these values are used.